src/coprocessor/dag/expr/builtin_cast.rs

```diff
@@ -97,15 +97,41 @@ impl FnCall {
     }
 
     pub fn cast_time_as_int(&self, ctx: &StatementContext, row: &[Datum]) -> Result<Option<i64>> {
-        unimplemented!()
+        let val = try!(self.children[0].eval_time(ctx, row));
+        if val.is_none() {
+            return Ok(None);
+        }
+        let mut val = val.unwrap().into_owned();
+        try!(val.round_frac(mysql::DEFAULT_FSP));
+        let dec = try!(val.to_decimal());
+        let res = try!(convert::dec_to_i64(dec));
+        Ok(Some(res))
     }
 
     pub fn cast_duration_as_int(
         &self,
         ctx: &StatementContext,
         row: &[Datum],
     ) -> Result<Option<i64>> {
-        unimplemented!()
+        let val = try!(self.children[0].eval_duration(ctx, row));
+        if val.is_none() {
+            return Ok(None);
+        }
+        let mut val = val.unwrap().into_owned();
+        try!(val.round_frac(mysql::DEFAULT_FSP));
+        let dec = try!(val.to_decimal());
```

Comment:
Can it be converted to a decimal then round? So that you don't have to clone the duration.